Explorer/Assets/Scripts/SceneRunner/Scene/ISceneFacade.cs

```diff
@@ -9,6 +9,8 @@ namespace SceneRunner.Scene
     public interface ISceneFacade : IUniTaskAsyncDisposable, IDisposable
     {
         SceneShortInfo Info { get; }
+        ISceneStateProvider? SceneStateProvider { get; }
```

Comment:
I don't think it can be `null`. So you don't really need `?` and check for nullity in systems

Reply:
>I've already had problems with EmptySceneFacade

Maybe introduce a flag in the facade then to be more explicit?

Reply:
I just re-tested with `main` latest changes, after merging that into this branch, and there are currently 2 problems:
1. When the world loads, the scene that the main player is in doesn't have the `sceneFacade.SceneStateProvider.IsCurrent` in `true`. That's unrelated to this PR and I can "bypass" that problem by re-entering the scene by walk.
2. Empty scenes don't have any `SceneStateProvider` but I think that's OK since we don't really need that stuff instantiated for those, right? that's why the `ISceneFacade.SceneStateProvider` can be `null`.

By introducing a Flag do you mean a flag for empty scenes? something like `IsEmpty` ?